app/helpers/schema_helper.py

```diff
@@ -35,7 +35,7 @@ def is_first_block_id_for_group(cls, survey_json, group_id, block_id):
 
     @staticmethod
     def get_last_block_id(survey_json):
-        return survey_json['groups'][0]['blocks'][-1]['id']
+        return survey_json['groups'][-1]['blocks'][-1]['id']
```

Comment:
It's used by get_last_location which get's the last group id. The test schema it was working on initially only had one group but I added another so it highlighted the flaw.